File: 프로그래머스/0/181931. 등차수열의 특정한 항만 더하기/등차수열의 특정한 항만 더하기.py
def solution(a, d, included):
    answer = [a]
    sum = 0
    
    for i in range(len(included)-1):
        a += d
        answer.append(a)

    for i in range(0, len(included)):
        if included[i] == True:
            sum += answer[i]
        elif included[i] == False:
            continue
    
    return sum

File: 프로그래머스/0/181931. 등차수열의 특정한 항만 더하기/등차수열의 특정한 항만 더하기.java
class Solution {
    public int solution(int a, int d, boolean[] included) {
        
        int sum = a;
        int total = 0;
        
        for(int i=0; i<included.length; i++){
            if(included[i] == true){
                total += sum;
                sum += d;
            }
            else if(included[i] == false){
                sum += d;
            }
        }
        
        return total;
    }
}